### PR TITLE
Migration of ca-rating web component

### PR DESCRIPTION
**Pull request by ElementForge**

### CODE REVIEW

File: components/ca-rating.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+         * @description Executes when any attribute is changed on the element
+         * @param {string} attrName - the name of the attribute to have changed
+         * @param {string} oldVal - the old value of the attribute
+         * @param {string} newVal - the new value of the attribute
+         * @returns {undefined}
+         */
+        attributeChangedCallback(attrName, oldVal, newVal) {
+            // only need to re-render the component if the attribute value has changed
+            if (oldVal !== newVal) {
+                this[attrName] = newVal;
```
Comment (norisalsaadie) [inline components/ca-rating.js]: review if needed in stage 1 refactor. Might have been needed due to polyfill issues.

File: components/ca-rating.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+        /** @property {string} ca-rating.defaults - gets the defaults of the element */
+        get defaults() {
+            return {
+                from: 1,   // the lowest ranking value to be displayed
+                to: 10,    // the highest ranking value to be displayed
+                value: -1  // the initially selected ranking value
+            };
+        }
+
```
Comment (norisalsaadie) [inline components/ca-rating.js]: The following attributes were not migrated:
attached: https://bitbucket.org/carepo/weblab-common-ui/src/b4002a5dd55bdc1e19a7b08cbf4bc6a7e12bb301/components/ca-rating.js?at=master&fileviewer=file-view-default#ca-rating.js-71

selectedItem: https://bitbucket.org/carepo/weblab-common-ui/src/b4002a5dd55bdc1e19a7b08cbf4bc6a7e12bb301/components/ca-rating.js?at=master&fileviewer=file-view-default#ca-rating.js-80

File: components/ca-rating.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    './helpers/create-element.js',
+    './helpers/cancel-event.js',
+    'document-register-element'
+], (clearElement, createElement, cancelEvent) => {
+    /**
+     * @exports ca-rating
+     * @description A custom HTML element (Web Component) that can be created using
+     * document.createElement('ca-rating') or included in a HTML page as an element.
+     */
+    class Rating extends HTMLElement {
```
Comment (kevmarchant) [inline components/ca-rating.js]: I would call it CaRating for consistency

File: components/ca-rating.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        }
+
+        /**
+          * @description Internal method to handles child interaction events
+          * @access private
+          * @type {Event}
+          * @param {Event} e - the element that the event happened on
+         * @returns {undefined} nothing
+          */
+        eventDelegate(e = event) {
```
Comment (kevmarchant) [inline components/ca-rating.js]: we're removing eventDelegate in the next phase I assume?

File: test/ca-rating-test.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    </template>
+</test-fixture>
+<script>
+    suite('<ca-rating>', () => {
+        let caRating;
+
+        // TODO: CREATE TESTS, there were none associated with the original component
+        setup(done => {
+            SystemJS.import('ca-rating.js').then(() => {
+                caRatingn = fixture('ca-rating');
```
Comment (kevmarchant) [inline test/ca-rating-test.html]: small typo

Comment (kevmarchant) [inline test/ca-rating-test.html]: didn't the linting pull this up as a non-defined variable?

Comment (norisalsaadie) [inline test/ca-rating-test.html]: Linting isn't enabled on the .html files yet...

Comment (norisalsaadie) [inline test/ca-rating-test.html]: I think I'll bypass the process and fix this myself, good catch.

**Pull request merged**